puppet-mode.el

```diff
@@ -1109,6 +1117,69 @@ for each entry."
 ;;;###autoload
 (add-to-list 'auto-mode-alist '("\\.pp\\'" . puppet-mode))
 
+;; this are only the mostly used puppet code keywords
+;; ( sorted by the order they are used in a puppet class )
+(setq puppet-kwdList
+      '(
+       ;; header
+       "class" "define"
+       "include" "inherits"
+       ;; types
+       "file" "package"
+       "cron" "exec"
+       "user" "sshkey"
+       "tidy" "notify"
+       "group" "service"
+       ;; options
+       "ensure" "name" "owner" "group"
+       "mode" "require" "notify" "hasrestart"
+       "hasstatus" "enable" "path" "home"
+       "password" "message" "withpath" "provider"
+       "comment" "managehome" "membership" "command"
+       "hour" "minute" "weekday" "month"
+       "monthday" "recurse" "content" "target"
+       "alias" "audit" "before" "loglevel"
+       "noop" "notify" "require" "schedule"
+       "stage" "subscribe" "tag"
+       ;; values
+       "present" "absent"
+       "running" "stopped"
+       "installed" "true"
+       "false" "undef"
+       ))
+
+(add-to-list 'completion-at-point-functions 'puppet-complete-symbol)
```

Comment:
No global changes to completion functions!  We don't want `puppet-complete-symbol` in any other mode.

Add the function in the body of `puppet-mode`, with `add-hook`, and be sure to pass a non-nil value for the `LOCAL` argument, so that the completion function is set up only for the current buffer.  

Take a look at `python-mode` (`M-x find-function RET python-mode`), for instance, to see how completion functions are set up.